### DISCUSSION TITLE
Slack Plugin: Failed to send report to Slack: RequestError: self-signed certificate in certificate chain

### DISCUSSION THREAD

**Discussion by priya-dsp** [artilleryio/artillery#3248] [Artillery]
Hi @hassy , I have integrated slack plugin with Artillery and able to view slack messages when I run the script via local PC however, when we run script via remote PC/VM, slack plugin is not able to send report to slack channel. error is - Slack Plugin: Failed to send report to Slack: RequestError: self-signed certificate in certificate chain. 

I tried both --insecure flag and tls in the config, but I'm still getting this error. Can you suggest ?

**Comment by hassy** [comment]
Very strange! This could be due to a corporate proxy or a self-signed certificate in the trust list on the VM. You can try setting the `NODE_TLS_REJECT_UNAUTHORIZED` environment variable to `0` in the VM (with something like `export NODE_TLS_REJECT_UNAUTHORIZED=0`) to see if that fixes it. This is not something we'd usually recommend due to potential security implications, but if you're running Artillery on a trusted VM the risk is somewhat reduced.

**Reply by hassy** [reply]
(`--insecure` doesn't work because it only affects requests made by Artillery's HTTP engine, whereas the Slack plugin is calling the Slack API directly)
